### DISCUSSION TITLE
issue when change the network

### DISCUSSION THREAD

**Discussion by RaveenaClarisco** [rainbow-me/rainbowkit#2034] [Feedback]
I recently integrated the RainbowKit wallet into my DApp, including support for both the Ethereum and Polygon networks. However, I'm encountering an issue where, upon connecting the wallet in my DApp, it defaults to the Ethereum network. Even when I switch to other networks like Sepolia or Binance (BNB) within my MetaMask extension, the DApp still displays the wrong network. What's even more frustrating is that upon refreshing the DApp, it reverts back to the Ethereum network, despite my MetaMask being set to the BNB network. I've checked my React application code but couldn't find any hardcoded chain IDs that might be causing this discrepancy. This inconsistency between MetaMask and the DApp's RainbowKit integration is a significant bug that needs addressing. Could you kindly investigate and provide feedback on this matter? You can reach me at suryagowtham567@gmail.com. Thank you for your attention to this issue.

**Comment by 0xmkh** [comment]
@RaveenaClarisco Apologies for the inconvenience. We have a bug ticket [here](https://github.com/rainbow-me/rainbowkit/issues/2038) regarding MetaMask defaulting to random chain and not to the current connected chain after refresh. 

For the switch chain bug, are you also able to provide a small reproducible example ?
